sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp

```diff
@@ -6192,9 +6192,9 @@ namespace Azure { namespace Storage { namespace Blobs {
         request.AddHeader("x-ms-version", c_ApiVersion);
         if (options.Timeout.HasValue())
         {
-          request.AddQueryParameter("timeout", std::to_string(options.Timeout.GetValue()));
+          request.GetUrl().AppendQuery("timeout", std::to_string(options.Timeout.GetValue()));
         }
-        request.AddQueryParameter("comp", "lease");
+        request.GetUrl().AppendQuery("comp", "lease");
```

Comment:
Is it expected for the caller to keep calling `GetUrl()` before any URL modifications, or is the expected coding pattern that the user calls `request.GetUrl()` once, stores it in a local, and then keeps calling `AppendX()`, until done?

Reply:
Either way is okay. As long as the local is a reference.